vivisect/parsers/elf.py

```diff
@@ -84,18 +84,20 @@ def makeRelocTable(vw, va, maxva, addbase, baseaddr):
     'amd64':'sysvamd64call',
 }
 
-def loadElfIntoWorkspace(vw, elf, filename=None):
+def loadElfIntoWorkspace(vw, elf, filename=None, arch=None, platform=None, filefmt='elf'):
```

Comment:
these parameters seem like something the elf object should contain/identify ( and potentially allow you to override ).